raiden_contracts/deploy/contract_verifier.py

```diff
@@ -199,6 +200,16 @@ def _verify_deployed_contract(
         # Check that the deployed bytecode matches the precompiled data
         blockchain_bytecode = self.web3.eth.get_code(contract_instance.address).hex()
         compiled_bytecode = self.contract_manager.get_runtime_hexcode(contract_name)
+
+        if contract_name == CONTRACT_TOKEN_NETWORK_REGISTRY:
+            # We need to link the libs into the contract bytecode.
+            # As this is run in the tests in fake file systems, do poor mans linking here
+            lib_address = contracts[LIBRARY_TOKEN_NETWORK_UTILS]["address"]
+            normalized_address = lib_address[2:].lower()
+            compiled_bytecode = compiled_bytecode.replace(
+                "__$34600480520cb524a2c423e33a5b4dd437$__", normalized_address
```

Comment:
Does this need to be replaced by a dynamic value in a followup, or is this some kind of magic constant?

Reply:
So this is the hash of the library path, I guess adding the calculation (if only for future reference) makes sense.
```py
 token_network_libs_key = "data/source/lib/TokenNetworkUtils.sol:TokenNetworkUtils"
"__${}$__".format(encode_hex(keccak(bytes(token_network_libs_key, 'utf-8')))[2:36])
> '__$34600480520cb524a2c423e33a5b4dd437$__'
```

I'm not really happy on how the linking is done here, but it's the best solution I could find. The problem is that `compile_contracts` doesn't now the address of the library yet, so it cannot do the linking. Here I'd rather not encode the relations between the contracts, but I don't see way around it.

Reply:
Yes, adding that calculation would make this both easier to understand and adapt to changes.